src/Simplify_Shuffle.cpp

```diff
@@ -238,6 +238,20 @@ Expr Simplify::visit(const Shuffle *op, ExprInfo *bounds) {
         }
     }
 
+    // Try to collapse a slice of slice.
+    if (op->is_slice() && (new_vectors.size() == 1)) {
+        if (const Shuffle *inner_shuffle = new_vectors[0].as<Shuffle>()) {
+            if (inner_shuffle->is_slice() && (inner_shuffle->vectors.size() == 1)) {
```

Comment:
Do you need the inner shuffles to be one vector? This might be useful for slices of concats too?

Reply:
Hmm, I was going to extend to any number of vectors, but it seems that:

a) make_slice only accepts a single vector.
b) is_slice doesn't actually check that vectors.size() == 1 and only checks that indices form ramp.

I can add an overload for make_slice which accepts multiple vectors, but wanted to check in case there are maybe assumptions somewhere that slice can only take a single vector (in which case, probably might be a good idea to add a check into is_slice).

Reply:
and seems that slice of concats is basically the same as a slice of multiple vectors?

Reply:
I don't think we need a new helper, but it seems like writing something like `make_slice(make_slice(make_concat(...), ...), ...)` would hit this case.

Reply:
As discussed offline, the simplifier doesn't do what I thought it did here :)